binary/type.proto

```diff
@@ -38,145 +46,68 @@ message Type {
-        Variation variation = 1;
-        Nullability nullability = 2;
-    }
-
-    message FP32 {
-        Variation variation = 1;
-        Nullability nullability = 2;
-    }
-
-    message FP64 {
-        Variation variation = 1;
-        Nullability nullability = 2;
-    }
+    // The type's nullability
+    Nullability nullability = 33;
+
+    // The type's variation. Optional.
+    Variation variation = 34;
+
+    message Boolean {}
+    message I8 {}
+    message I16 {}
+    message I32 {}
+    message I64 {}
+    message FP32 {}
+    message FP64 {}
+    message String {}
+    message Binary {}
+    message Timestamp {}
+    message Date {}
+    message Time {}
+    message TimestampTZ {}
+    message IntervalYear {}
+    message IntervalDay {}
+    message UUID {}
 
-    message String {
-        Variation variation = 1;
-        Nullability nullability = 2;
-    }
-
-    message Binary {
-        Variation variation = 1;
-        Nullability nullability = 2;
-    }
-
-    message Timestamp {
-        Variation variation = 1;
-        Nullability nullability = 2;
-    }
-
-    message Date {
-        Variation variation = 1;
-        Nullability nullability = 2;
-    }
-
-    message Time {
-        Variation variation = 1;
-        Nullability nullability = 2;
-    }
-
-    message TimestampTZ {
-        Variation variation = 1;
-        Nullability nullability = 2;
-    }
-
-    message IntervalYear {
-        Variation variation = 1;
-        Nullability nullability = 2;
-    }
-
-    message IntervalDay {
-        Variation variation = 1;
-        Nullability nullability = 2;
-    }
-
-    message UUID {
-        Variation variation = 1;
-        Nullability nullability = 2;
-    }
-
-    // Start compound types.
     message FixedChar {
         int32 length = 1;
-        Variation variation = 2;
-        Nullability nullability = 3;
     }
 
     message VarChar {
         int32 length = 1;
-        Variation variation = 2;
-        Nullability nullability = 3;
     }
 
     message FixedBinary {
         int32 length = 1;
-        Variation variation = 2;
-        Nullability nullability = 3;
     }
 
     message Decimal {
         int32 scale = 1;
         int32 precision = 2;
-        Variation variation = 3;
-        Nullability nullability = 4;
     }
 
     message Struct {
         repeated Type types = 1;
-        Variation variation = 2;
-        Nullability nullability = 3;
-    }
-
-    message NamedStruct {
-        // list of names in dfs order
-        repeated string names = 1;
-        Struct struct = 2;
     }
 
     message List {
         Type type = 1;
-        Variation variation = 2;
-        Nullability nullability = 3;
     }
 
     message Map {
         Type key = 1;
         Type value = 2;
-        Variation variation = 3;
-        Nullability nullability = 4;
     }
-
-    message Variation {
-        int32 organization = 1;
-        string name = 2;
-    }
-
 }
 
+message NamedStruct {
```

Comment:
Since `NamedStruct` wasn't a type in the `oneof` (and wasn't nullable or variable), there's no utility in keeping it embedded in the `Type` message.